insights/client/phase/v1.py

```diff
@@ -207,8 +207,12 @@ def post_update(client, config):
     # put this first to avoid conflicts with register
     if config.unregister:
         if reg_check:
-            logger.info('Unregistration is not currently available.')
-            sys.exit(constants.sig_kill_bad)
+            logger.info('Unregistering this host with Insights.')
+            if client.unregister():
+                get_scheduler(config).remove_scheduling()
```

Comment:
IMO this should be in client.handle_unregistration() for cleanliness parity with the old function. I can revisit if you disagree.

Reply:
So I was originally planning to handle it in handle_unregistration(), but I noticed that the daily scheduling was started in v1(https://github.com/RedHatInsights/insights-core/blob/unregister_host_delete/insights/client/phase/v1.py#L179) so I included it here to be consistent with register. I don't have a strong preference either way though so I don't mind changing it around.